ddtrace/contrib/botocore/patch.py

```diff
@@ -52,23 +69,155 @@ def inject_trace_data_to_message_attributes(trace_data, entry):
 
 
-def inject_trace_to_sqs_batch_message(args, span):
+def inject_trace_to_sqs_or_sns_batch_message(args, span):
+    # type: (List[Any], Span) -> None
+    """
+    :args: contains args for the current botocore action, SNS/SQS records are at index 1
+    :span: the span which provides the trace context to be propagated
+
+    Inject trace headers into MessageAttributes for all SQS or SNS records inside a batch
+    """
     trace_data = {}
     HTTPPropagator.inject(span.context, trace_data)
     params = args[1]
 
-    for entry in params["Entries"]:
+    # An entry here is an SNS or SQS record, and depending on how it was published,
+    # it could either show up under Entries (in case of PutRecords),
+    # or PublishBatchRequestEntries (in case of PublishBatch).
+    entries = params.get("Entries", params.get("PublishBatchRequestEntries", []))
+    for entry in entries:
         inject_trace_data_to_message_attributes(trace_data, entry)
 
 
-def inject_trace_to_sqs_message(args, span):
+def inject_trace_to_sqs_or_sns_message(args, span):
+    # type: (List[Any], Span) -> None
+    """
+    :args: contains args for the current botocore action, SNS/SQS record is at index 1
+    :span: the span which provides the trace context to be propagated
+
+    Inject trace headers into MessageAttributes for the SQS or SNS record
+    """
     trace_data = {}
     HTTPPropagator.inject(span.context, trace_data)
     params = args[1]
 
     inject_trace_data_to_message_attributes(trace_data, params)
 
 
+def inject_trace_to_eventbridge_detail(args, span):
+    # type: (List[Any], Span) -> None
+    """
+    :args: contains args for the current botocore action, EventBridge records are at index 1
+    :span: the span which provides the trace context to be propagated
+
+    Inject trace headers into the EventBridge record if the record's Detail object contains a JSON string
+    Max size per event is 256KB (https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-putevent-size.html)
+    """
+    params = args[1]
+    if "Entries" not in params:
+        log.debug("Unable to inject context. The Event Bridge event had no Entries.")
+        return
+
+    for entry in params["Entries"]:
+        detail = {}
+        if "Detail" in entry:
+            try:
+                detail = json.loads(entry["Detail"])
+            except ValueError:
+                log.debug("Detail is not a valid JSON string")
+                continue
+
+        detail["_datadog"] = {}
+        HTTPPropagator.inject(span.context, detail["_datadog"])
+        detail_json = json.dumps(detail)
+
+        # check if detail size will exceed max size with headers
+        detail_size = len(detail_json)
+        if detail_size >= MAX_EVENTBRIDGE_DETAIL_SIZE:
+            log.debug("Detail with trace injection exceeds max allowed size")
+            continue
+
+        entry["Detail"] = detail_json
+
+
+def get_kinesis_data_object(data):
```

Comment:
I'd rather have this function `raise` rather than return `None` if it fails to decode any data. The exception would be caught by the broader exception handler we have around all of the injection calls